optuna/samplers/grid.py

```diff
@@ -0,0 +1,178 @@
+import collections
+import itertools
+import random
+
+from optuna.samplers.base import BaseSampler
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Union
+
+    from optuna.distributions import BaseDistribution  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
+    from optuna.study import Study  # NOQA
+
+    GridValueType = Union[str, float, int, bool, None]
+
+
+class GridSampler(BaseSampler):
+    """Sampler using grid search.
+
+    With :class:`~optuna.samplers.GridSampler`, the trials suggest all combinations of parameters
+    in the given search space during the study.
+
+    This sampler is based on *relative sampling*.
+    See also :class:`~optuna.samplers.BaseSampler` for more details of 'relative sampling'.
```

Comment:
I mind this comment causes a misunderstanding because the implementation is actually not based on relative sampling.

Reply:
Thanks for your comment.

Right; I put those sentences just because the sampler uses `sample_relative`, but they may cause confusion..

I removed the sentences.